server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -779,4 +801,13 @@ private static CircuitBreaker inFlightRequestsBreaker(CircuitBreakerService circ
         // We always obtain a fresh breaker to reflect changes to the breaker configuration.
         return circuitBreakerService.getBreaker(CircuitBreaker.IN_FLIGHT_REQUESTS);
     }
+
+    @AccessLevel(Access.PUBLIC)
```

Comment:
My concern with this Annotation based approach is that it only means something if `rest.enforce_api_protections` is enabled, which it won't be by default.
I think that is going to be confusing to people who read the code without that (relatively esoteric) piece of information.

When we spoke @rjernst suggested we name the annotation `@ServerlessAccessLevel` or something like that. That goes a long way to solving the confusion, but leaks "serverless" into the code, where it isn't actually needed.
I'd love to find a way to minimize the confusion that doesn't require us to inject "serverless" into the annotation  name.

Reply:
I have not thought of a way around this. Is this a showstopper for doing this in the code (vs an external list)? Regardless of what we do in the code, it's going to be meaningless outside of Serverless.

Reply:
I'm not that concerned about leaking "serverless". It's a known thing, and it's a conscious choice. This is specifically for serverless, so calling it another name would be more confusing IMO.